src/components/IstioCertsInfo/IstioCertsInfo.tsx

```diff
@@ -0,0 +1,181 @@
+import {
+  Button,
+  Card,
+  CardBody,
+  CardHeader,
+  Grid,
+  GridItem,
+  Modal,
+  Title,
+  Tooltip,
+  TooltipPosition
+} from '@patternfly/react-core';
+import { KialiAppState } from 'store/Store';
+import { istioCertsInfoSelector, lastRefreshAtSelector } from 'store/Selectors';
+import { ThunkDispatch } from 'redux-thunk';
+import { KialiAppAction } from 'actions/KialiAppAction';
+import { bindActionCreators } from 'redux';
+import { IstioCertsInfoActions } from 'actions/IstioCertsInfoActions';
+import { connect } from 'react-redux';
+import { TimeInMilliseconds } from 'types/Common';
+import { CertsInfo } from 'types/CertsInfo';
+import { PFColors } from 'components/Pf/PfColors';
+import { KialiIcon } from 'config/KialiIcon';
+import { infoStyle } from 'styles/DropdownStyles';
+
+type IstioCertsInfoState = {
+  showModal: boolean;
+  certsError: Boolean;
+};
+
+type IstioCertsInfoProps = {
+  lastRefreshAt: TimeInMilliseconds;
+  setIstioCertsInfo: (istioCertsInfo: CertsInfo[]) => void;
+  certsInfo: CertsInfo[];
+  ref: React.RefObject<any>;
+};
+
+class IstioCertsInfo extends React.Component<IstioCertsInfoProps, IstioCertsInfoState> {
+  constructor(props: IstioCertsInfoProps) {
+    super(props);
+    this.state = { showModal: false, certsError: false };
+  }
+
+  open = () => {
+    this.setState({ showModal: true });
+  };
+
+  close = () => {
+    this.setState({ showModal: false });
+  };
+
+  componentDidMount() {
+    this.fetchStatus();
+  }
+
+  componentDidUpdate(prevProps: Readonly<IstioCertsInfoProps>): void {
+    if (this.props.lastRefreshAt !== prevProps.lastRefreshAt) {
+      this.fetchStatus();
+    }
+  }
+
+  fetchStatus = () => {
+    API.getIstioCertsInfo()
+      .then(response => {
+        this.props.setIstioCertsInfo(response.data);
+        this.setState({ certsError: false });
+      })
+      .catch(_error => {
+        this.setState({ certsError: true });
+      });
+  };
+
+  showCertInfo = (certInfo: CertsInfo): JSX.Element => {
+    return (
+      <Grid>
+        <GridItem span={3}>
+          <b>Issuer</b>
+        </GridItem>
+        <GridItem span={9}>{certInfo.issuer}</GridItem>
+        <GridItem span={3}>
+          <b>Valid from</b>
+        </GridItem>
+        <GridItem span={9}>{certInfo.notBefore}</GridItem>
+        <GridItem span={3}>
+          <b>Valid until</b>
+        </GridItem>
+        <GridItem span={9}>{certInfo.notAfter}</GridItem>
+        {certInfo.dnsNames && (
+          <>
+            <GridItem span={3}>
+              <b>DNS Names</b>
+            </GridItem>
+            <GridItem span={9}>
+              {certInfo.dnsNames && certInfo.dnsNames.map((dnsName, index) => <li key={index}>{dnsName}</li>)}
+            </GridItem>
+          </>
+        )}
+      </Grid>
+    );
+  };
+
+  render() {
+    return (
+      <Modal
+        isSmall={true}
+        isOpen={this.state.showModal}
+        onClose={this.close}
+        title="Certicates information"
```

Comment:
Typo - Certicates should be certificates :)

Reply:
@abonas thanks! it's fixed now.